### PR TITLE
Initial MNLogit work

### PR DESCRIPTION
**Pull request by jbrockmendel**
To start off on the right foot, this is mostly just flake8 cleanup in/around MNLogit and MultinomialResults.

Small but nonzero overlap with #2197

### CODE REVIEW

**Comment by josef-pkt** [comment]
This pull request fixes 1 alert - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-d8930c0747986f664089aa53b8c018b5d01256c2)

**fixed alerts:**

* 1 for First argument to super() is not enclosing class

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by josef-pkt** [comment]
This pull request fixes 1 alert - [view on lgtm.com](https://lgtm.com/projects/g/statsmodels/statsmodels/rev/pr-cf269f4b5069bcbd993618ee89ad3be5976a61db)

**fixed alerts:**

* 1 for First argument to super() is not enclosing class

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by jbrockmendel** [comment]
@josef-pkt  mnlogit is a priority right?

File: statsmodels/discrete/discrete_model.py [hunk 1]
```diff
@@ -3893,45 +3910,88 @@ def __init__(self, model, bnryfit):
...
 
-    def _get_endog_name(self, yname, yname_list, all=False):
+    def __init__(self, model, mlefit):
+        super(MultinomialResults, self).__init__(model, mlefit)
+        self.J = model.J
+        self.K = model.K
+        self.nobs = model.nobs
+
+    @deprecate_kwarg('all', 'use_all')
+    def _get_endog_name(self, yname, yname_list, use_all=False):
```
Comment (josef-pkt) [inline statsmodels/discrete/discrete_model.py]: we don't need to deprecate changes in private, internal functions or methods

**Comment by josef-pkt** [comment]
> mnlogit is a priority right?
true but it's high priority for 0.10, at least as far as API changes are affected.

AFAICS, confint wrapper is a change in substance and API and needs unit tests, and if we would merge that before the release it would have to be clear that it is the public facing solution that will last.

**Comment by jbrockmendel** [comment]
> true but it's high priority for 0.10,

I've been trying to bother you less since realizing it's still pre-release season.  (BTW That politician you dislike also said something about delegating maintenance being a sign of weakness.)

> we don't need to deprecate changes in private, internal functions or methods

Excellent.  So quickie PR that just fixes that one thing?

> confint wrapper is a change in substance and API [...] it would have to be clear that it is the public facing solution that will last.

Right now `conf_int` doesn't work _at all_.  This is not necessarily the best solution, but it _a_ solution.  We can put a note in the docstring and issue an all-caps warning that it's experimental and subject to change.  Corner cases get found and rough edges get sanded via lots of use, and that's never going to happen as long as MNLogitResults is effectively unusable.

**Comment by jbrockmendel** [comment]
This is clearly not going anywhere.  If at some point you get serious about fixing MNLogit let me know and I'll volunteer to review.

**Pull request closed without merging**